From 579d625f017b70140048f88d4f2136236bcd873c Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Wed, 28 Oct 2026 20:06:15 +0900
Subject: [PATCH 08/15] sparrow-hawk-fw: Add license information

The recipe used LICENSE = "CLOSED", so the license of the PCIe firmware
was not described in the SPDX data or in the license output, although
LICENCE.r8a779g_pcie_phy is installed with the firmware.

Use the license file fetched from linux-firmware as the license text of
"Firmware-r8a779g_pcie_phy". Set S to WORKDIR because the sources are
single files without an archive.

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 recipes-kernel/linux/sparrow-hawk-fw.bb | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/recipes-kernel/linux/sparrow-hawk-fw.bb b/recipes-kernel/linux/sparrow-hawk-fw.bb
index d7badbf..746e719 100644
--- a/recipes-kernel/linux/sparrow-hawk-fw.bb
+++ b/recipes-kernel/linux/sparrow-hawk-fw.bb
@@ -1,5 +1,7 @@
 SUMMARY = "Binary firmware for Sparrow hawk board"
-LICENSE = "CLOSED"
+LICENSE = "Firmware-r8a779g_pcie_phy"
+LIC_FILES_CHKSUM = "file://LICENCE.r8a779g_pcie_phy;md5=0b20e76a9a004b83c4a1c87e2153bbad"
+NO_GENERIC_LICENSE[Firmware-r8a779g_pcie_phy] = "LICENCE.r8a779g_pcie_phy"
 PR = "r0"
 
 PACKAGE_ARCH = "${MACHINE_ARCH}"
@@ -13,6 +15,8 @@ SRC_URI:append:sparrow-hawk = " \
     ${PCIE_FIRMWARE_LIC} \
 "
 
+S = "${WORKDIR}"
+
 do_compile[noexec] = "1"
 FILES:${PN}:sparrow-hawk += "/usr/lib/firmware/"
 
